fn_grpc_interface/fn_grpc_interface/components/function_grpc.py

```diff
@@ -0,0 +1,290 @@
+import sys
+import os
+import json
+import grpc
+import re
+import inspect
+from google.protobuf.json_format import MessageToDict
+
+
+class FunctionComponent(ResilientComponent):
+    """Component that implements generic wrapper for the gRPC client on the Resilient platform"""
+
+    def __init__(self, opts):
+        """constructor provides access to the configuration options"""
+        super(FunctionComponent, self).__init__(opts)
+        self.options = opts.get("fn_grpc_interface", {})
+
+    @handler("reload")
+    def _reload(self, event, opts):
+        """Configuration options have changed, save new values"""
+        self.options = opts.get("fn_grpc_interface", {})
+
+    def _get_interface_file_names(self, base_dir=None, files_dir=None):
+        """
+        component that return the absolute path of all files in the given directory
+        :param base_dir:  interface pb2 files parent directory
+        :param files_dir:  interface pb2 files directory i.e same as package name
+        :return: lists of all pb2 files absolute path inside the given directory path
+        """
+        __dir_path = os.path.join(base_dir, files_dir)
+        return os.listdir(__dir_path)
+
+    def _get_grpc_interface_module(self, interface_files=[], base_dir=None, files_dir=None):
+        """
+        component that imports all the modules in the current workspace and returns the given imported modules objects.
+        :param interface_files: List of all buffer pb2 files absolute path to import in the current work space
+        :param base_dir:  interface pb2 files parent directory.
+        :param files_dir: interface pb2 files directory i.e same as package name.
+        :return: returns imported interface pb2 files module objects on the successful execution, else returns None object.
+        """
+        assert isinstance(interface_files, list), "This Parameter "'"interface_files"'" should be list data type"
+        assert base_dir is not None, "The param base_dir should contain parents parent directory of gRPC interface \
+        files"
+        assert files_dir is not None, "The param files_dir should points to parent directory of gRPC interface files"
+
+        if interface_files:
+            abs_file_path_json = dict()
+            interface_module = []
+
+            # creating files abs path
+            for file_name in interface_files:
+                abs_file_path = os.path.join(base_dir, files_dir, file_name)
+                abs_file_path_json[file_name] = abs_file_path
+
+            # Adding Module path into sys.path variable
+            interface_dir_path = os.path.join(base_dir, files_dir)
+            if interface_dir_path not in sys.path:
+                sys.path.append(interface_dir_path)
+
+            # compatibility for python 2.7 and 3.6
+            logging.info("Loading The Modules : {}".format(interface_files))
+
+            try:
+                import importlib.util
```

Comment:
* We should keep all imports at the top of the file
* You could put the following at the top of the file
```
import six
#PY2
try:
   from importlib.util import spec_from_file_location, module_from_spec
#PY3
expect ImportError:
  from imp import load_source
```

* Then in the `_get_grpc_interface_module` function we can make use of the `six` library to check if we are using Python 2 or 3
```
        if six.PY2:
            #use spec_from_file_location and module_from_spec

        elif six.PY3:
            #use load_source

        else:
            raise ValueError("We do not support this version of Python")
```

Reply:
ok